### ISSUE TITLE
No prompt after closing editor when piping command that doesn't terminate

### ISSUE THREAD

**Issue by roblourens** [microsoft/vscode#203166]
Testing #202921

I noticed a difference in behavior. Not sure whether it's an issue. If I have a command that doesn't terminate, and I use `--wait`, then I don't get a prompt after closing the editor until I ctrl+C.

1. `top | code-insiders --wait -`
2. Close the editor that opens
3. In Insiders, I don't get back to the prompt until I ctrl+C in the terminal. But in stable, I do get back to the prompt at this point.

**Comment by bpasero** [comment]
Yeah this is a good catch and it takes me a bit to understand what we should be doing here actually. I think:
* when `--wait` is used, we should return prompt when files close (as we used to)
* without `--wait` at least we should return prompt when the Code process terminates

**Comment by bpasero** [comment]
I decided to do 1) but not 2).
